src/Common/src/CoreLib/System/Text/DecoderReplacementFallback.cs

```diff
@@ -19,6 +22,12 @@ public DecoderReplacementFallback() : this("?")
         {
         }
 
+#if MONO
+        internal DecoderReplacementFallback(SerializationInfo info, StreamingContext context) => _strDefault = info.GetString("strDefault");
```

Comment:
We probably also need to read from `_strDefault` for more recent Mono versions, right?

Reply:
yeah probably makes sense to handle it as a fallback